llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -8962,8 +9108,16 @@ LoopVectorizationPlanner::tryToBuildVPlanWithVPRecipes(VFRange &Range) {
           Legal->isInvariantAddressOfReduction(SI->getPointerOperand()))
         continue;
 
-      VPRecipeBase *Recipe =
-          RecipeBuilder.tryToCreateWidenRecipe(Instr, Operands, Range, VPBB);
+      VPRecipeBase *Recipe = nullptr;
+
+      PartialReductionChain Chain;
+      if (getInstructionsPartialReduction(Instr, Chain))
+        Recipe =
+            RecipeBuilder.tryToCreatePartialReduction(Range, Chain, Operands);
```

Comment:
I think this creation attempt should be moved to `VPRecipeBuilder::tryToWiden` in the Instruction::Add case, with a fallthrough to the normal add case if we don't find a matching partial reduction.

Reply:
That requires changing the base class of `VPPartialReductionRecipe` to `VPWidenRecipe` which I'll hold off on doing for now.